### PR TITLE
Cleverwordcloud function definition and unit tests

### PR DESCRIPTION
**Pull request by Vikiano**
I have completed the function definitions, edge cases exception tests and unit test cases. Please review.

### CODE REVIEW

**Comment by adrianne-l** [comment]
There is a conflicting file man/CleverWordCloud.Rd. Need to resolve conflicts before merging PR.

**Comment by ajuliettet** [comment]
> There is a conflicting file man/CleverWordCloud.Rd. Need to resolve conflicts before merging PR.

@Vikiano is supposed to be the only one that makes and pushes this file because his function is CleverWordCloud.Rd right? It was somehow there when I did get pull. I think Victor's man/CleverWordCloud.Rd is the only one we need.

File: R/CleverWordCloud.R [hunk 1]
```diff
@@ -1,15 +1,42 @@
...
+#' @param data DataFrame containing the words and their frequencies
 #'
 #' @return An image
 #' @export
 #'
 #' @examples
-#' CleverWordCloud(sample_text, max_words=200)
-CleverWordCloud <- function(text, stopwords=STOPWORDS, shape=CLOUD, max_words=100) {
+#' CleverWordCloud(c("cry", "foot", "maximum"))
+CleverWordCloud <- function(cleverlemstem_output) {
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]: I think the variable should have a more general name. People don't need to use the output of our other functions. 

```suggestion
CleverWordCloud <- function(text) {
```

File: R/CleverWordCloud.R [hunk 2]
```diff
@@ -1,15 +1,42 @@
...
+  library(wordcloud)
+  library(wordcloud2)
+  library(RColorBrewer)
+  library(tm)
+  library(htmlwidgets)
+  library(testthat)
+  library(usethis)
+  library(tidyverse)
 
+  if(is.data.frame(cleverlemstem_output)){
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
  if(is.data.frame(text)){
```

File: R/CleverWordCloud.R [hunk 3]
```diff
@@ -1,15 +1,42 @@
...
+    stop("Cannot return wordcloud for dataframe object, only vector of characters are allowed as input")
+  }
+  if(is.numeric(cleverlemstem_output)){
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
  if(is.numeric(text)){
```

File: R/CleverWordCloud.R [hunk 4]
```diff
@@ -1,15 +1,42 @@
...
+    stop("Cannot return wordcloud for numeric object, only vector of characters are allowed as input")
+  }
+  if(!is.character(cleverlemstem_output)){
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
  if(!is.character(text)){
```

File: R/CleverWordCloud.R [hunk 5]
```diff
@@ -1,15 +1,42 @@
...
+    stop("Cannot return wordcloud for non-character input")
+  }
+  if(is.integer(cleverlemstem_output)){
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
  if(is.integer(text)){
```

File: R/CleverWordCloud.R [hunk 6]
```diff
@@ -1,15 +1,42 @@
...
+    stop("Cannot return wordcloud for integer object, only vector of characters are allowed as input")
+  }
+
+
+  dataframe <- data.frame(word=cleverlemstem_output) |>
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
  dataframe <- data.frame(word=text) |>
```

File: R/CleverWordCloud.R [hunk 7]
```diff
@@ -1,15 +1,42 @@
 #' This function generates a visually appealing word cloud with customized shape and stopwords.
 #'
-#' @param text Input an array of text / strings.
-#' @param stopwords Input an set of strings.
-#' @param shape Input a string that represents the path to an image.
-#' @param max_words Input an integer to indicate the maximum number of words included in the word cloud
+#' @param data DataFrame containing the words and their frequencies
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
#' @param text Input a character vector
```

File: R/CleverWordCloud.R [hunk 8]
```diff
@@ -1,15 +1,42 @@
 #' This function generates a visually appealing word cloud with customized shape and stopwords.
```
Comment (ajuliettet) [inline R/CleverWordCloud.R]:
```suggestion
#' This function generates a meaningful word cloud with customized stopwords
```

**Comment by Vikiano** [comment]
I just changed the input argument name to "text"
